lang/python/wiredtiger/init.py

```diff
@@ -47,6 +47,23 @@
 me = sys.modules[__name__]
 sys.path.append(os.path.dirname(__file__))
 
+# Find the sanitizer environment variable.
+# FIXME-WT-TBD: Rename these testutil flags
+if os.environ.get("TESTUTIL_TSAN") == "1":
+    import subprocess
+    # FIXME - can we assume that the tsan lib is always libtsan.so.0?
+    command = "/opt/mongodbtoolchain/v4/bin/clang --print-file-name libtsan.so.0"
+    tsan_so_path = subprocess.run(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True).stdout.strip()
+
+    # This is a bit hacky but if LD_PRELOAD is already set then python has the required libraries.
+    # Clear it now so that ./wt doesn't get affected by it.
+    if os.environ.get("LD_PRELOAD") is None:
+        os.environ["LD_PRELOAD"] = tsan_so_path
+        python = sys.executable
+        os.execl(python, python, *sys.argv)
+    else:
+        os.environ["LD_PRELOAD"] = ""
```

Comment:
This will be an issue if we need `LD_PRELOAD` for anything other than TSan. 
Right now that's only TCMalloc which isn't compatible with sanitisers, but for safety we should strip the sanitiser path rather than overriding LD_PRELOAD. 

Similarly for `os.environ["LD_PRELOAD"] = tsan_so_path` above

Reply:
f0397dab4

Reply:
I took a shot at refactoring in [2b6389e](https://github.com/wiredtiger/wiredtiger/pull/10812/commits/2b6389e8b3018ecec46f666e34212c5abe5b92a7)